src/compiler/checker.ts

```diff
@@ -6816,7 +6876,10 @@ namespace ts {
                             const statement = factory.createVariableStatement(/*modifiers*/ undefined, factory.createVariableDeclarationList([
                                 factory.createVariableDeclaration(varName, /*exclamationToken*/ undefined, serializeTypeForDeclaration(context, typeToSerialize, symbol, enclosingDeclaration, includePrivateSymbol, bundled))
                             ], NodeFlags.Const));
-                            addResult(statement, name === varName ? ModifierFlags.Export : ModifierFlags.None);
+                            addResult(statement,
+                                target && target.flags & SymbolFlags.Property && target.escapedName === InternalSymbolName.ExportEquals ? ModifierFlags.Ambient

```

Comment:
Care to add an inline comment explaining why a `Property` symbol needs a `declare` modifier (and not `export`)?

Reply:
Yep, it's because JSON types exported with `[module.]export=` will already emit an `export = blah`, so the inlined JSON type itself doesn't need `export`, but `declare`.